packages/amesos2/src/Amesos2_Cholmod_TypeMap.hpp

```
@@ -68,92 +65,13 @@
-      ret.complexpair[1] = Teuchos::as<float>(t.imag());
-      return( ret );
-    }
-
-    static Amesos2::CHOL::complex safeConvert( const TypeFrom t )
-    {                           // adapt conversion as necessary
-      Amesos2::CHOL::complex ret;
-      //ret.r = Teuchos::as<float>(t.real());
-      //ret.i = Teuchos::as<float>(t.imag());
-      ret.complexpair[0] = Teuchos::as<float>(t.real());
-      ret.complexpair[1] = Teuchos::as<float>(t.imag());
-      return( ret );
-    }
-  };
-
-  // Also convert *from* New_Solver types
-  template <typename TypeTo>
-  class ValueTypeConversionTraits<TypeTo, Amesos2::CHOL::complex>
-  {
-  public:
-    static TypeTo convert( const Amesos2::CHOL::complex t )
-    {                           // adapt conversion as necessary
-      typedef typename TypeTo::value_type value_type;
-      //value_type ret_r = Teuchos::as<value_type>( t.real );
-      //value_type ret_i = Teuchos::as<value_type>( t.imag );
-      value_type ret_r = Teuchos::as<value_type>(t.complexpair[0]);
-      value_type ret_i = Teuchos::as<value_type>(t.complexpair[1]);
-      return ( TypeTo( ret_r, ret_i ) );
-    }
-
-    static TypeTo safeConvert( const Amesos2::CHOL::complex t )
-    {                           // adapt conversion as necessary
-      typedef typename TypeTo::value_type value_type;
-      //value_type ret_r = Teuchos::as<value_type>( t.real );
-      //value_type ret_i = Teuchos::as<value_type>( t.imag );
-      value_type ret_r = Teuchos::as<value_type>(t.complexpair[0]);
-      value_type ret_i = Teuchos::as<value_type>(t.complexpair[1]);
-      return ( TypeTo( ret_r, ret_i ) );
-    }
-  };
-
-  //@}  End Conversion group
-
-} // end namespace Teuchos
-
-
 namespace Amesos2 {
 
   // forward declaration due to circular reference
   template <class, class> class Cholmod;
 
   template <>
-  struct TypeMap<Cholmod,float>
+  struct TypeMap<Cholmod,float> // Cholmod does not support float yet
```

@MicheldeMessieres what happens if someone tries to call this solver with float enabled, will it throw an error with a warning, silently convert to double, or silently fail?
Similar question for complex_float cases.

---

This will give a run time error that float is not supported.